### PR TITLE
Convert hibernate-3.3 tests from groovy to java

### PR DESCRIPTION
**Pull request by jaydeluca**
Related to #7195

### CODE REVIEW

File: instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          Session session = sessionFactory.openSession();
+          session.beginTransaction();
+          try {
+            parameter.sessionMethodTest.accept(session, prepopulated.get(0));
+          } catch (RuntimeException e) {
+            // We expected this, we should see the error field set on the span.
```
Comment (breedx-splk) [inline instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java:40]: I know this was just brought over as-is, but if the exception really is expected then there should probably be a `fail()` call in the try block.

File: instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+  void testHibernateReplicate(Parameter parameter) {
+    testing.runWithSpan(
+        "parent",
+        () -> {
+          Session session = sessionFactory.openSession();
+          session.beginTransaction();
+          try {
+            parameter.sessionMethodTest.accept(session, prepopulated.get(0));
+          } catch (RuntimeException e) {
+            // We expected this, we should see the error field set on the span.
```
Comment (breedx-splk) [inline instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java:121]: Same comment about the `fail()` or remove the comment.

**Review by breedx-splk — APPROVED** [review]
Thanks! Not the easiest review. :) Looks to have the same coverage to me, and I appreciate eliminating some of the duplication that exists in the groovy implementation.

File: instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java (hunk 1 above)
Comment (jaydeluca) [reply instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java]: Thank you for the review. So only one of the tests actually threw an exception (`testHibernateFailedReplicate`), so I removed the try/catch from all others, and added an additional event assertion for the exception in the one test where it was applicable. If this wasn't what you had in mind just let me know and I can revisit.

Comment (breedx-splk) [reply instrumentation/hibernate/hibernate-3.3/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/hibernate/v3_3/SessionTest.java]: Nah that's perfect, thanks! Seems like just leftover copypasta, so I appreciate you cleaning it up.

**Comment by breedx-splk** [comment]
@opentelemetrybot license

**Pull request merged**